oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandlerTest.java

```diff
@@ -357,6 +286,174 @@ void shouldRedirectToFrontendErrorPageIfUserInfoRequestIsUnsuccessful()
+
+        @BeforeAll
+        static void init() {
+            when(authorizationService.validateRequest(any(), any())).thenReturn(true);
+        }
+
+        @BeforeEach
+        void setup() throws UnsuccessfulCredentialResponseException {
+            mockedIdentityHelper = mockStatic(IdentityHelper.class);
+            when(IdentityHelper.identityRequired(anyMap(), anyBoolean(), anyBoolean()))
+                    .thenReturn(false);
+            when(tokenService.sendTokenRequest(any())).thenReturn(SUCCESSFUL_TOKEN_RESPONSE);
+            when(tokenService.sendUserInfoDataRequest(any(HTTPRequest.class)))
+                    .thenReturn(USER_INFO);
+            usingValidSession();
+            usingValidClientSession();
+            usingValidClient();
+        }
+
+        @AfterEach
+        void tearDown() {
+            mockedIdentityHelper.close();
+        }
+
+        @Test
+        void shouldPerformBackChannelLogoutWhenAccountStatusIsBlocked() {
+            AccountInterventionStatus accountStatus =
+                    new AccountInterventionStatus(true, false, false, false);
+            when(configurationService.isAccountInterventionServiceEnabled()).thenReturn(true);
+            when(configurationService.isAccountInterventionServiceAuditEnabled()).thenReturn(true);
+            when(accountInterventionService.getAccountStatus(any())).thenReturn(accountStatus);
+
+            var event = new APIGatewayProxyRequestEvent();
+            setValidHeadersAndQueryParameters(event);
+
+            handler.handleRequest(event, null);
+
+            verify(logoutService)
+                    .handleAccountInterventionLogout(
+                            session,
+                            event,
+                            Optional.of(CLIENT_ID.toString()),
+                            Optional.of(SESSION_ID.toString()),
+                            accountStatus);
+        }
+
+        @Test
+        void shouldPerformBackChannelLogoutWhenResetPasswordIsRequired() {
+            AccountInterventionStatus accountStatus =
+                    new AccountInterventionStatus(false, true, false, true);
+            when(configurationService.isAccountInterventionServiceEnabled()).thenReturn(true);
+            when(configurationService.isAccountInterventionServiceAuditEnabled()).thenReturn(true);
+            when(accountInterventionService.getAccountStatus(any())).thenReturn(accountStatus);
+
+            var event = new APIGatewayProxyRequestEvent();
+            setValidHeadersAndQueryParameters(event);
+
+            handler.handleRequest(event, null);
+
+            verify(logoutService)
+                    .handleAccountInterventionLogout(
+                            session,
+                            event,
+                            Optional.of(CLIENT_ID.toString()),
+                            Optional.of(SESSION_ID.toString()),
+                            accountStatus);
+        }
+
+        @Test
```

Comment:
Could we add a test for the opposite of this? `shouldRedirectToIPVWhenIdentityRequiredAndAccountStatusIsSuspended`?

Reply:
Done and added a couple more while I was there